sdk/src/logger.rs

```diff
@@ -68,3 +72,21 @@ impl Logger {
         error!(target: "gadget", "[{}] {msg}", &self.id);
     }
 }
+
+/// Sets up the logging for any crate
+pub fn setup_log() {
+    std::panic::set_hook(Box::new(|info| {
+        error!(target: "gadget", "Panic occurred: {:#?}", info);
+    }));
+
+    setup_log_no_panic_hook()
+}
+
+fn setup_log_no_panic_hook() {
+    let _ = SubscriberBuilder::default()
+        .without_time()
+        .with_span_events(FmtSpan::NONE)
+        .with_env_filter(EnvFilter::from_default_env())
+        .finish()
+        .try_init();
+}
```

Comment:
Note this: https://github.com/webb-tools/gadget/pull/246#discussion_r1759134224

Something like this shouldn't really be in here. It's a loss of information in that you're setting a new panic hook and making assumptions about the setup of the tracing subscriber.

Reply:
I think what @Serial-ATA is saying is still valid. Isn't getting a full backtrace going to be important for people other than yourself? If this is reducing the standard error information that is presented I think this is bad.